pages/api/policies/validate.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2021 The Rode Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { StatusCodes, ReasonPhrases } from "http-status-codes";
+import fetch from "node-fetch";
+import { getRodeUrl } from "pages/api/utils/api-utils";
+
+const ALLOWED_METHODS = ["POST"];
+
+export default async (req, res) => {
+  if (!ALLOWED_METHODS.includes(req.method)) {
+    return res
+      .status(StatusCodes.METHOD_NOT_ALLOWED)
+      .json({ error: ReasonPhrases.METHOD_NOT_ALLOWED });
+  }
+
+  const rodeUrl = getRodeUrl();
+
+  try {
+    const policy = req.body;
+
+    const response = await fetch(`${rodeUrl}/v1alpha1/policies:validate`, {
+      method: "POST",
+      body: policy,
+    });
+
+    if (!response.ok) {
+      console.error(`Unsuccessful response from Rode: ${response.status}`);
+
+      const parsedResponse = await response.json();
+
+      if (
+        parsedResponse?.message?.includes("failed to compile") ||
```

Comment:
if you need me to make a field for this in the next PR i can @pickjasmine . Instead of searching the string in the response

Reply:
I think the current implementation of this is fine. Either way I'm going to have to parse the response and look for some field (or the value of a field in this case) so I wouldn't worry about it too much unless you think the error message we send back will change.